File: ThreadsApp/ThreadsApp/View/Chat/Cells/Switch/SwitchCell.m
//
//  SwitchCell.m
//  ThreadsApp
//
//  Created by Tabriz Dzhavadov on 13/07/2017.
//  Copyright © 2017 Brooma Service. All rights reserved.
//

#import "SwitchCell.h"

@interface SwitchCell()

@property (weak, nonatomic) IBOutlet UILabel *placeholderLabel;

@property (weak, nonatomic) IBOutlet UISwitch *switcher;

@end

@implementation SwitchCell

#pragma mark - Initialize

- (void) awakeFromNib {
    [super awakeFromNib];
    self.selectionStyle = UITableViewCellSelectionStyleNone;
}

#pragma mark - Configuring

- (void) construct {
    self.placeholderLabel.text = [CellTypeHelper textByType: self.type];
}

- (IBAction) switched: (UISwitch *) sender {
    if (self.delegate && [self.delegate respondsToSelector: @selector(switchCell:switched:)]) {
        [self.delegate switchCell: self switched: self.switcher.isOn];
    }
}

@end

File: ThreadsApp/ThreadsAppSwift/View/Chat/Cells/SwitchCell.swift
//
//  SwitchCell.swift
//  THRAppSwift-Polarbear
//
//  Created by Brooma Service on 16/05/2019.
//  Copyright © 2019 Brooma Service. All rights reserved.
//

import Foundation
import UIKit

protocol SwitchCellDelegate: class {
    func switchCell(_ cell: SwitchCell, switched: Bool)
}

class SwitchCell : UITableViewCell {
    
    weak var delegate: SwitchCellDelegate?
    
    var type : CellType = .DebugMode
    
    @IBOutlet weak var placeholderLabel: UILabel!
    
    @IBOutlet weak var switcher: UISwitch!
    
    // MARK: - Initialize
    
    override func awakeFromNib() {
        super.awakeFromNib()
        self.selectionStyle = .none
    }
    
    //MARK: - Configuring

    func construct() {
        self.placeholderLabel.text = CellTypeHelper.getCellText(forType: self.type)
    }
    
    @IBAction func switched(_ sender: UISwitch) {
        delegate?.switchCell(self, switched: switcher.isOn)
    }
}
